### DISCUSSION TITLE
Beginner question, upload GVRET to EVTVDue microcontroller

### DISCUSSION THREAD

**Discussion by ProgrammersCat** [collin80/SavvyCAN#566] [Q&A]
Hello all, 
I am trying to upload GVRET to the EVTVDue board. When compiling (Arduino IDE 2.0.3 on Win10) I get error messages about missing *.h files. Now the file unistd.h is missing. Which of the many different versions do I need to use and where do I store it? On Windows I don't have an **asm** directory.

Error message:
*****************************************
...\Arduino\libraries\SysCall/SysCall.h:25:10: fatal error: **asm**/unistd.h: No such file or directory
 #include <asm/unistd.h>
          ^~~~~~~~~~~~~~
compilation terminated.
exit status 1
Compilation error: exit status 1
********************************************

**Comment by ProgrammersCat** [comment]
Found something. I probably used a wrong SysCall.d file. I will download it and the additional missing files from 
https://github.com/collin80/SdFat-beta/tree/master/SdFat/src

**Comment by ProgrammersCat** [comment]
I am at a loss here. Where must "BaseBlockDriver.h" be copied to? I did it with 
..\Arduino\libraries\BaseBlockDriver
but got the following error message:

..\Arduino\libraries\BlockDriver/BlockDriver.h:26:10: fatal error: FatLib/BaseBlockDriver.h: No such file or directory
 #include "FatLib/BaseBlockDriver.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
exit status 1
Compilation error: exit status 1

**Comment by collin80** [comment]
Are you sure you're compiling for the Arduino Due board? I'm not aware of any "SysCall" library that should be in your libraries folder.  It's super easy to get into the weeds when installing libraries. I've been there myself. Sometimes you have to delete everything and start over because the mess is too large. 

Lastly, the best advice is to see if you really need to compile it yourself. If you plan to make changes then, by all means, get set up to compile it and I'll answer any questions that pop up. But, if you don't plan to change it then go to www.savvycan.com and download the GVRET updater which is just a binary flash tool. This will install the already compiled program for you.

**Comment by ProgrammersCat** [comment]
Hello Collin, thank you for your answer!
SavvyCan is now working with the binary flash board.
Before I compile my own code, I want to be able to compile your version. 
I have downloaded 'GVRET-PC.zip' from: 
https://store.evtv.me/products/evtvdue2?_pos=3&_sid=e04f934ae&_ss=r loaded
I got 'SdFat.h' from https://github.com/collin80/SdFat-beta/blob/master/SdFat/src/SdFat.h
'SdFat.h' line 26 requires SysCall.h 
Did I use a wrong SdFat.h version?
Thanks and best regards

**Comment by collin80** [comment]
Wow, you're right. The library in my repo doesn't really work to compile GVRET. I suppose that shows you how long it has been since I actually tried to compile it. In fact, I did not even have a copy of SdFat installed anywhere that GVRET could find it. The truth is that the version on savvycan.com is probably the last time I really compiled it. And, sure enough, I can't find a SysCall.h file anywhere in the Due files. The root cause of all this is that different versions of SdFat end up being included with different board files. I had moved on to ESP32 and Teensy 4.1 development so I had to delete my copy of SdFat so they'd use theirs and work properly. So, anyway, it sure is broken. 

If you use Greiman's current version of SdFat it will compile: https://github.com/greiman/SdFat

I just tested that to work. You still need due_wire, can_common, due_can,  Wire_EEPROM, and mcp2515 libraries as well.

**Comment by ProgrammersCat** [comment]
Thanks again Collin
I still have many errors when compiling. Also in the SdFat version of Greiman, SysCall.h, and other linked libraries are called (e.g. FsBlockDeviceInterface.h) One library asks for the next (seemingly infinitely;)). Is it correct that I need to download these libraries individually? I'm just asking to make sure I'm not doing anything unnecessary. Maybe I am on the wrong way or there is an easier way.
Thanks and best regards

**Comment by ProgrammersCat** [comment]
Yeaa! I have discovered the green '<>Code' button! download ZIP function rocks!
Get on with it!

**Comment by ProgrammersCat** [comment]
Bingo! Compile and upload work now!

**Comment by collin80** [comment]
Congrats! Yes, it's difficult for anyone to get all the pieces together and get it compiling at first. There's a lot of libraries to get installed and working. After it's all set up you just have to resist the urge to switch machines and start over. Though, with experience comes an easier time in getting the pieces together.
